apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -178,6 +178,8 @@ export class UpsertWorkflowUseCase {
         controlsDto: step.controlValues,
       });
 
+      const updateStepId = existingStep?.stepId;
+      const syncToEnvironmentCreateStepId = step.stepId;
```

Comment:
we want to **store** the stepId from the step we got, it is valid for cases where we sync to the environment.